WordPress/Classes/ViewRelated/Tools/PromptViewController.swift

```diff
@@ -0,0 +1,123 @@
+import Foundation
+
+
+/// Protocol that defines the methods required by PromptViewController's Children(s).
+///
+@objc
+protocol PresentedViewController
+{
+    var doneButtonEnabled : Bool { get }
+    func cancelButtonWasPressed(sender: AnyObject)
+    func doneButtonWasPressed(sender: AnyObject)
+}
+
+
+/// ViewController container, that presents a Done / Cancel button, and forwards their events to
+/// the childrenViewController (which *must* implement PresentedViewController).
+///
+class PromptViewController : UIViewController
+{
+    deinit {
+        stopListeningToProperties(childrenViewController)
+    }
+    
+    init(viewController: UIViewController) {
+        super.init(nibName: nil, bundle: nil)
+        assert(viewController.conformsToProtocol(PresentedViewController))
```

Comment:
This makes me 😢 but I think we're stuck with it until all the potential targets are written in Swift. In that case we could just do this

```swift
    init<Controller: UIViewController where Controller: PresentedViewController>(viewController: Controller) {
```

I've tried doing this and then implementing an unsafe wrapper for ObjC, but I don't know how to force cast something into a class that conforms to a protocol, or if it's even possible.